cirq/google/engine/engine_job.py

```diff
@@ -285,6 +285,10 @@ def results(self) -> List[study.TrialResult]:
                   result_type == 'cirq.api.google.v2.Result'):
                 v2_parsed_result = v2.result_pb2.Result.FromString(result.value)
                 self._results = self._get_job_results_v2(v2_parsed_result)
+            elif result_type == 'cirq.google.api.v2.BatchResult':
```

Comment:
Could use the built-in Any mechanism for checking types here:
```python
elif result.Is(v2.batch_pb2.BatchResult.DESCRIPTOR):
    ...
```
We could change other cases as well though we'd have to leave in code to handle old `cirq.api.google` types if we still need that.

Reply:
Done.  I will leave the other cases how they are for now.